src/dds_simple.h

```diff
@@ -244,120 +244,178 @@ class DDSSimpleManager : public DDSManager
     // This function waits until it finds one (or more) Subscriber of topic T OR secondsToWait seconds expires.
     // Useful when making sure your messages are being Published on a certain topic.
     template <class T>
-    bool WaitForSubscriber(int secondsToWait = 2)
+    bool WaitForSubscriber(std::chrono::milliseconds timeToWait = std::chrono::seconds(2))
+    {
+        return GetNumberOfSubscribers<T>(1, timeToWait) > 0;
+    }
+
+    //Call WaitForPublisher(0) if you have already been discovered and want to see if you've lost all connection to publishers.
+    // readerName is not required unless user specifies a readerName when creating their Subscriber / Callback
+    template <class T>
+    bool WaitForPublisher(std::chrono::milliseconds timeToWait = std::chrono::seconds(2), std::string readerName = "")
     {
+        return GetNumberOfPublishers<T>(1, timeToWait, readerName) > 0;
+    }
+
+    // Function that will wait until [max_wait] passes or until we find [min_count] number of Subscribers, whichever is faster
+    template<class T>
+    int GetNumberOfSubscribers(int min_count, std::chrono::milliseconds max_wait = std::chrono::seconds(2))
+    {
+        const std::chrono::milliseconds waitIncriment(100);
+        std::chrono::milliseconds timeWaited(0);
+        auto startTime = std::chrono::system_clock::now();
```

Comment:
Since this is a relative wait, you probably want to use std::steady_clock to avoid being impacted by stuff like NTP time adjustments, etc.